### PR TITLE
feat: [CDS-72816]: do not allow container step group configuration for CI stage

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- Based on stage type, if it is CI, do not show `Enable container based execution` switch

#### Screenshots
Screenshot 2023-06-23 at 4 57 20 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by YounixIO** [comment]
retrigger cypress-rest

**Comment by YounixIO** [comment]
retrigger cypress-cd

**Comment by YounixIO** [comment]
retrigger cypress-pipeline

**Comment by YounixIO** [comment]
retrigger cypress-cv

**Comment by YounixIO** [comment]
retrigger cypress-rest

File: src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/StepGroupStepEdit.tsx [hunk 1]
```diff
@@ -26,10 +26,12 @@ import type { StepViewType, StepFormikFowardRef } from '@pipeline/components/Abs
 import { setFormikRef } from '@pipeline/components/AbstractSteps/Step'
 import { VolumesTypes } from '@pipeline/components/Volumes/Volumes'
 import { OsTypes } from '@pipeline/utils/constants'
+import { StageType } from '@pipeline/utils/stageHelpers'
 import type { MultiTypeListType, MultiTypeListUIType } from '../StepsTypes'
 import { getNameAndIdentifierSchema } from '../StepsValidateUtils'
 import { KubernetesStepGroupInfra } from './KubernetesStepGroupInfra'
 import type { K8sDirectInfraStepGroupElementConfig, StepGroupFormikValues, TolerationFormik } from './StepGroupUtil'
+import { StepGroupCustomStepProps } from './StepGroupStep'
```
Comment (litvinovstas) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/StepGroupStepEdit.tsx:34]: import type.

**Review by litvinovstas — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/StepGroupStepEdit.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/StepGroupStep/StepGroupStepEdit.tsx:34]: Will check this in next PR. Thanks.

**Review by ishantmehtaharness — APPROVED** [review]
looks good chetan. test cases looks good too

**Pull request merged**